src/js/popup.js

```diff
@@ -77,11 +90,29 @@ document.addEventListener('DOMContentLoaded', () => {
     })
   }
 
+  // Listen for click events in the sidebar button
+  // Hide if not if Firefox
+  var sidebarButton = document.getElementById('open-sidebar')
+  if (sidebarButton) {
+    if (isFirefox) {
+      document.getElementById('open-sidebar').addEventListener('click', () => {
+        chrome.sidebarAction.open()
+        window.close()
+      })
+    } else {
+      sidebarButton.style.display = 'none'
+    }
+  }
+
   // Listen for click events in the settings
   document.getElementById('settings').addEventListener('click', () => {
     if (chrome.runtime.openOptionsPage) {
       // New way to open options pages, if supported (Chrome 42+).
       chrome.runtime.openOptionsPage()
+      const isPopup = document.body.classList.contains('popup-page')
+      if (isPopup) {
+        window.close()
+      }
```

Comment:
What does this do? I think it closes the pop up window if one clicks on settings, is that correct?

Reply:
I see what what this is doing now. It only closes the pop up window, it doesn't close the side bar. Nice.